Tools/autotest/common.py

```diff
@@ -196,7 +196,10 @@ def fetch_parameters(self):
     def reboot_sitl(self):
         """Reboot SITL instance and wait it to reconnect."""
         self.mavproxy.send("reboot\n")
-        self.mavproxy.expect("tilt alignment complete")
+        self.mav.wait_heartbeat() # makes sure we have a mav_type
+        # ArduSub doesn't do tilt alignment after boot
+        if self.mav.mav_type != mavutil.mavlink.MAV_TYPE_SUBMARINE:
```

Comment:
intersting ! I am pretty sure it was working yesterday when I test it ....

Reply:
ok it doesn't work on my patch, so it was surely working I rebase ..

Reply:
the reboot is working fine on master, what error did you get ?

Reply:
I get this testing ArduSub on master:
```
....
Init ArduSub V3.6-dev (820f9bd2)

Free RAM: 131072
load_all took 0us
0 0 0 validate_structures:348: Validating structures
DataFlash_File: buffer size=16384

Init completeReceived 850 parameters
Saved 833 parameters to test.ArduSub/logs/2018-10-12/flight1/mav.parm
APM: EKF2 IMU0 Origin set to GPS
APM: EKF2 IMU1 Origin set to GPS
bind port 5762 for 2
Serial port 2 on TCP port 5762
bind port 5763 for 3
Serial port 3 on TCP port 5763
TSYS01 device is null!APM: MYGCS: 255, heartbeat lost
APM: EKF2 IMU1 is using GPS
APM: EKF2 IMU0 is using GPS
APM: MYGCS: 255, heartbeat lost
APM: MYGCS: 255, heartbeat lost
APM: MYGCS: 255, heartbeat lost
APM: MYGCS: 255, heartbeat lost
APM: MYGCS: 255, heartbeat lost
APM: MYGCS: 255, heartbeat lost
APM: MYGCS: 255, heartbeat lost
APM: MYGCS: 255, heartbeat lost
APM: MYGCS: 255, heartbeat lost
APM: MYGCS: 255, heartbeat lost
Flight battery warning
APM: MYGCS: 255, heartbeat lost
Timed out looking for tilt alignment complete
...
```

Reply:
@Williangalvani That log doesn't show anything from SITL after the reboot, so either it didn't reboot  or it took too long to reboot. Neither of which means the message isn't there. That message comes from the EKF, I don't see how it wouldn't be present on Sub.

Reply:
From my tests I saw this sometime, the error comes generaly from the expect that isn't capable to catch the right text.
@oxinarf, the message from sitl are here! the instance reboot well but the expect thing failed... we need to change that